modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/notification/recipient/RoleNotificationRecipientBuilder.java

```diff
@@ -129,6 +137,25 @@ protected void deactivate() {
 	@Reference
 	protected UserLocalService userLocalService;
 
+	private boolean _allowAncestorSites() {
+		if (FeatureFlagManagerUtil.isEnabled("LPD-23210")) {
+			try {
+				WorkflowDefinitionConfiguration
+					workflowDefinitionConfiguration =
+						configurationProvider.getSystemConfiguration(
+							WorkflowDefinitionConfiguration.class);
+
+				return !workflowDefinitionConfiguration.
+					preventNotifyingAncestorSites();
+			}
+			catch (ConfigurationException configurationException) {
+				throw new RuntimeException(configurationException);
+			}
+		}
```

Comment:
Hey @p-albuquerque keep the same name as the configuration here private boolean `_preventNotifyingAncestorSites()`

Thanks!

Reply:
Hey @p-albuquerque do you mind use this approach [here](https://github.com/liferay/liferay-portal/blob/81ff7798b6baa38454455c94e86e72b1520bea6b/modules/apps/portal-security/portal-security-script-management-web/src/main/java/com/liferay/portal/security/script/management/web/internal/configuration/helper/ScriptManagementConfigurationHelperImpl.java#L113-L117) instead of using configurationProvider.getSystemConfiguration, because I'm not sure how to handle all situations with this approach, I don't know if it will throw a NPE when the configuration is not defined for example, the other approach was being used long time ago for us and it's still good enough when compared to this approach.

the example that I've shared before, you only have to create an active method that should be annotated with @Active and @Modified and you have to create a volatile variable for this configuration in this class.

Thank you!

Reply:
Hey @p-albuquerque use the return early approach:

````java
if (!FeatureFlagManagerUtil.isEnabled("LPD-23210")) {
  return true;
}

return _workflowDefinitionConfiguration.preventNotifyingAncestorSites();
````

Thanks!

Reply:
I had tried the activate method approach, but with it I wasn't able to change the configuration value in runtime during the [integration test](https://github.com/liferay-workflow/liferay-portal/blob/6585ab31c7665e034d979121b0db195150db138b/modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java#L723-L729). It always still the same as I select in **setUpClass**
I also tried to change the value the same way the [setUpClass](https://github.com/liferay-workflow/liferay-portal/blob/6585ab31c7665e034d979121b0db195150db138b/modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java#L178-L184), but nothing changes

Reply:
I see now, the way I did this activate approach was slightly different than the example you sent.